server.h

```diff
@@ -51,8 +52,8 @@ class Server  {
 public:
 
     // Returns the instance of the server using the singleton pattern
-    static std::unique_ptr<Server>& GetInstance() {
-        static std::unique_ptr<Server> instance;
+    static Server* GetInstance() {
+        static Server* instance = (Server*) malloc(sizeof(Server));
```

Comment:
whenever status handler is called and GetInstance() is invoked, it mallocs something?